scripts/data/preferences/nectar.py

```diff
@@ -0,0 +1,123 @@
+from datasets import load_dataset, Dataset
+import random
+import argparse
+from tqdm import tqdm
+
+def clean_prompt(prompt):
+    if prompt.startswith("Human:"):
+        prompt = prompt[6:]
+    if prompt.endswith("Assistant:"):
+        prompt = prompt[:-10]
+    return prompt.strip()
+
+def is_valid_response(response):
+    return "Human:" not in response and "Assistant:" not in response
+
+def load_nectar_dataset(subset="lmsys-chat-1m", deduplication=False):
+    # Load the Nectar dataset
+    nectar_dataset = load_dataset("berkeley-nest/Nectar", split="train")
+    print(f"Original Nectar dataset size: {len(nectar_dataset)}")
+    
+    if deduplication:
+        print("Deduplication enabled. Loading UltraFeedback dataset...")
+        ultra_feedback = load_dataset("openbmb/UltraFeedback", split="train")
+        print(f"UltraFeedback dataset size: {len(ultra_feedback)}")
+        
+        # Create a set of UltraFeedback instructions for faster lookup
+        ultra_feedback_instructions = set(ultra_feedback['instruction'])
+
+        # Filter Nectar dataset, excluding lmsys-chat-1m and anthropic-hh
+        filtered_dataset = nectar_dataset.filter(lambda example: example['source'][0] not in ["lmsys-chat-1m", "anthropic-hh"])
```

Comment:
Why is this called `deduplication`? Would it be more appropriate to call it `--remove-sources "lmsys-chat-1m" "anthropic-hh"`?

Reply:
It's for deduplication of prompts found in UltraFeedback, our other popular dataset. There's no overlap with LMSYS + HH so we handle them differently.